programs/uxd/src/instructions/lsd/initialize_lsd_depository.rs

```diff
@@ -0,0 +1,153 @@
+    let depository_bump = *ctx
+        .bumps
+        .get("depository")
+        .ok_or_else(|| error!(UxdError::BumpError))?;
+    let profits_token_account_bump = *ctx
+        .bumps
+        .get("profits_token_account")
+        .ok_or_else(|| error!(UxdError::BumpError))?;
+
+    let redeemable_mint_unit = 10_u64
+        .checked_pow(
+            ctx.accounts
+                .controller
+                .load()?
+                .redeemable_mint_decimals
+                .into(),
+        )
+        .ok_or_else(|| error!(UxdError::MathError))?;
+
+    let depository = &mut ctx.accounts.depository.load_init()?;
+    depository.bump = depository_bump;
+    depository.version = LSD_DEPOSITORY_ACCOUNT_VERSION;
+    depository.collateral_mint = ctx.accounts.collateral_mint.key();
+    depository.collateral_mint_decimals = ctx.accounts.collateral_mint.decimals;
+    depository.liquidation_mint = ctx.accounts.liquidation_mint.key();
+    depository.liquidation_mint_decimals = ctx.accounts.liquidation_mint.decimals;
+    depository.profits_token_account = ctx.accounts.profits_token_account.key();
+    depository.profits_token_account_bump = profits_token_account_bump;
+
+    // Configuration
+    depository.borrowing_disabled = true;
+    depository.redeemable_amount_under_management_cap = DEFAULT_REDEEMABLE_UNDER_MANAGEMENT_CAP
+        .checked_mul(redeemable_mint_unit)
+        .ok_or_else(|| error!(UxdError::MathError))?;
+    depository.borrowing_fee_bps = 0;
+    depository.repay_fee_bps = 0;
+    depository.loan_to_value_bps = 0;
+    depository.liquidation_fee_bps = 0;
+    depository.profits_beneficiary = Pubkey::default();
+
+    // Accounting
+    depository.collateral_amount_deposits = u64::MIN;
+    depository.redeemable_amount_under_management = u64::MIN;
+
+    // Stats
+    depository.collateral_amount_liquidated = u128::MIN;
+    depository.borrow_fee_accrued = u128::MIN;
+    depository.repay_fee_accrued = u128::MIN;
+    depository.liquidation_fee_accrued = u128::MIN;
+    depository.profits_collected = u128::MIN;
+
+    emit!(InitializeLsdDepositoryEvent {
+        version: ctx.accounts.controller.load()?.version,
+        depository_version: depository.version,
+        controller: ctx.accounts.controller.key(),
+        depository: ctx.accounts.depository.key(),
+        collateral_mint: ctx.accounts.collateral_mint.key(),
+        liquidation_mint: ctx.accounts.liquidation_mint.key(),
+    });
+
+    Ok(())
+}
+
+// Validate input arguments
+impl<'info> InitializeLsdDepository<'info> {
+
+    pub(crate) fn validate(&self) -> Result<()> {
+        validate_is_program_frozen(self.controller.load()?)?;
+
+        // Collateral mint and redeemable mint should share the same decimals to justify their 1:1 swapping
+        require!(
+            self.collateral_mint
+                .decimals
+                .eq(&self.controller.load()?.redeemable_mint_decimals),
```

Comment:
mSOL and stSOL are 9 decimals

https://explorer.solana.com/address/mSoLzYCxHdYgdzU16g5QSh3i5K3z3KZK7ytfqcJm7So
https://explorer.solana.com/address/7dHbWXmci3dT8UFYWYZweBLXgycu7Y3iL6trKn1Y7ARj

Reply:
Oh yhea that's a copypasta gone wrong, didn't really go over it yet, but would have prob missed it! Thanks